### PR TITLE
Pages and UI update

### PR DESCRIPTION
**Pull request by jasonleyser**
- Added modal page states for the home, account and shortcut pages.
- Bug fixes
- Added slate logo back
- Added Inter font
- CSS variables
- Progress bar for account storage

**Pages**

Main modal
Screen Shot 2021-09-13 at 3 50 06 PM

Account
Screen Shot 2021-09-13 at 3 50 20 PM

Shortcuts
Screen Shot 2021-09-13 at 3 50 12 PM

### CODE REVIEW

**Comment by martinalong** [comment]
image
Looks like it's still misaligned on some websites such as https://medium.com/yom-ai/rest-api-with-node-js-and-elasticsearch-1368cf9df02a
Adding display: flex to .modalButtonSVG seemed to solve it for me

**Comment by martinalong** [comment]
We shouldn't be committing the build folder to git because it's a file that's compiled from the other files. Instead, add `build/` to the .gitignore file and delete the build files that you've already pushed to git, then push again
https://community.frontity.org/t/git-commit-for-build-directory/466/4

**Comment by martinalong** [comment]
Can you run npm install prettier and then go into every file and hit ctrl + s? This will auto format it with prettier, and help us avoid awkward formatting like this: 
image

**Comment by martinalong** [comment]
There's a few organizational standards that we use in the main app that would be good to use in the extension as well. Right now, they may not be significant since the extension is relatively small, but as it grows it'll help keep everything neat.

- Use @emotion/css 
    - first, `npm install @emotion/css`
    - then you can use it like we do in the regular app with import { css } from "@emotion/css". The purpose of this is to be able to use in page styles instead of having one giant css file that's hard to parse

Code reuse:
The purpose of putting repeated code or values into a shared file and then importing it is so that we can save on code, and so that when we need to make updates to values or code, we only have to do so in one place rather than having to hunt for every instance of it. Imagine trying to control + F all the places where we used a certain color in order to change it, versus changing one line in a contants file.
- put the svg.js file (along with any other shared "utilities") in a folder at the root level called `common` (aka common/svg.js, common/constants.js)
- you will probably also need a constants.js file in this common folder. It's up to you what you put in there, but the trick is to always be looking out for reused values that could belong in it. at the very least, put all the colors that you are using in there, like what we do in the main app
- you may also need a strings.js file in the common folder which is for commonly reused string manipulations (like getting a url from a cid), or a validitions.js which is for commonly reused checks (like checking if a username is valid). For examples, see the main app code

File: src/Components/Button.js [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
+require('typeface-inter');
 
 const Button = (props) => {
 
+	console.log('this is a button', props)
+
 	let svg;
-	if(props.icon == 'tag') {
-		svg = <Icons.Tag />
+	if(props.icon == 'command') {
```
Comment (martinalong) [inline src/Components/Button.js]: Instead of this complicated else if structure, you can just create an object matching names to svg's, and it'll also be faster (since looking up something in a hashtable--which is essentially what an object is--is faster than stepping through a bunch of if statements sequentially)

```
const icons = {
  bookmark: <Icons.Bookmark />,
  plus: <Icons.Plus />,
  ...
}

svg = icons[props.icon];
```

File: src/Components/Button.js [hunk 2]
```diff
@@ -1,23 +1,34 @@
 import React, { useState } from 'react';
```
Comment (martinalong) [inline src/Components/Button.js]: I find it neatest to organize imports by "style" and put a space in between the different styles, such as:
```
import React, { useState } from "react";

import * as Icons from "../Components/Icons";
import * as Strings from "/common/strings";

import classes from "../App.module.css";
```

File: src/Components/Button.js [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
 import * as Icons from '../Components/Icons';
 import classes from '../App.module.css';
+require('typeface-inter');
```
Comment (martinalong) [inline src/Components/Button.js]: Don't mix require and import syntax. They are equivalent, but import syntax is the new es6 standard, so this should be written as `import Inter from "typeface-inter"`

Comment (martinalong) [reply src/Components/Button.js]: However, I think here you should stick to using font faces instead of using a package specifically just to load this font.

You would do that by adding it as a global style with:
```
  @font-face {
    font-family: 'inter-regular';
    src: url('https://slate.textile.io/ipfs/bafkreic3hkcuwvrmf6trweqcjp62valsfjh3zvwacikoreynakgw67wrvy');
  }
```

Fonts can either be loaded from a local file (like the one you have in static) or from a url like the one above. I'd recommend loading from a url like we do in the main app bc then it can load from that url simultaneously while the rest of the app loads from a different server, rather than competing for the same "loading bandwidth" from a single source. This would mean you can delete the Inter-Regular.ttf font from the public folder

**Comment by martinalong** [comment]
Can you delete the unused logo.svg, tag.svg, and other svg files now that you're just exporting from the svg.js folder instead? Same goes for the unused logo images like icon-128.png:
image

**Comment by martinalong** [comment]
What are public/settings.html and public/settings.js for? Curious why they're in the public folder

File: src/Components/Button.js [hunk 4]
```diff
@@ -32,6 +43,12 @@ const Button = (props) => {
 				<div className={classes.modalButtonText}>
 					{props.text}
 				</div>
+				<div style={{ position: 'absolute', right: '32px', color: '#8E9093', fontSize: '14px' }}>
+					{props.command &&
+						<span className={classes.modalCommandIcon}>⌥</span> 
+					}
+					<span className={classes.modalKeyIcon}>{props.shortcut}</span>
```
Comment (martinalong) [inline src/Components/Button.js]: For the styling for this, instead of applying the background and border radius directly to the text, i'd recommend putting the text in a div and then playing around with height/width and padding to make it look more like the designs. The background is currently much more narrow than in the designs.

Implementation: 
image

The design:
image

**Comment by martinalong** [comment]
The extension has a weird bug on this page: https://developer.chrome.com/docs/extensions/reference/cookies/
image

**Comment by martinalong** [comment]
image
The slate symbol for when there isn't an image seems to have a thin line under it

**Comment by martinalong** [comment]
And on dropbox, the modal goes to one side:
image

**Comment by martinalong** [comment]
image
font sizing on the description is too large on substack

**Comment by martinalong** [comment]
> The extension has a weird bug on this page: https://developer.chrome.com/docs/extensions/reference/cookies/
> image

image
More image sizing issues on knownorigin.io

**Comment by martinalong** [comment]
On some dropbox links, it also seems to scroll with the page instead of being static: https://www.dropbox.com/sh/4mm6if0if9j01g4/AADMniHJGxsHmdRL_Smd1PFva?dl=0

File: src/Components/Button.js (hunk 3 above)
Comment (jasonleyser) [reply src/Components/Button.js]: for security reasons chrome doesn't allow linking to am external url for anything like css, scripts, fonts and all that.

**Comment by jasonleyser** [comment]
@martinalong 

The recent update to this PR includes:
- code cleanup (deleted old files in public and unused components)
- added a common folder in /src including: constants, strings, styles and svgs 
- added shadow dom and new css structure (emotion doesn't seem to be working so right now its in a constructable stylesheet, but im still working around this)
- ran prettier on all files
- added cookies and fetching api key with login session


https://github.com/slate-engineering/slate-web-extension/pull/4/commits/18351bf59565a43fd3af0b1bc84ce073a1a7d769

**Comment by martinalong** [comment]
image
The add to library button still seems misaligned. The text is too low compared to the icon

**Comment by martinalong** [comment]
image
The account icon also looks thinner than the rest and slightly to the right, likely because strokeWidth is not specified for it:
image

Also, no need to keep all the SVGs that the slate application has in the SVG.js file. Only the ones that the extension uses

**Comment by martinalong** [comment]
image
Sometimes the image shows as a broken image rather than the default slate logo when no image is found, like for this page: https://github.com/brianc/node-postgres/issues/442

**Comment by martinalong** [comment]
Also, a lot of pages which I have never saved show up with a green checkmark indicating that they have been saved. Can you track down the cause of this? And if it's the endpoint returning incorrect values, let me know and I'll fix it

I've definitely never saved this link to slate, but it shows a green check:  
image

**Comment by martinalong** [comment]
image

I'm getting this error when I tried to build it. Just a reminder to delete that unused import:
image

**Comment by martinalong** [comment]
image
On some websites the url is a really large font for some reason. This is the site: https://askanyc.com/about/

But then on others it's smaller https://www.exploretock.com/perse/:
image

**Comment by martinalong** [comment]
https://user-images.githubusercontent.com/33686587/135197456-5783663a-af5c-494c-9874-a29f89b2c749.mov

The loader spinner moves around as it spins. Take a look at how we do it in the main application. It may be b/c the SVG is not centered in the box that is rotating, or b/c SVGs have a bit of space underneath them if you don't do display: "block" (this happens b/c they are inline elements by default, and inline elements leave a bit of space underneath them for the parts of letters that go below the "baseline")

**Comment by martinalong** [comment]
The loader spinner moves around as it spins. Take a look at how we do it in the main application. It may be b/c the SVG is not centered in the box that is rotating, or b/c SVGs have a bit of space underneath them if you don't do display: "block" (this happens b/c they are inline elements by default, and inline elements leave a bit of space underneath them for the parts of letters that go below the "baseline"

I slacked you a video (didn't show up well on github)

**Comment by martinalong** [comment]
Saved links now don't seem to show up as saved (opposite problem of last time, where everything showed up as a green check mark)

image

**Comment by martinalong** [comment]
image
There's some extra space below the SVG here. I think this may be one where putting "display: block" on the svg will solve it

**Comment by martinalong** [comment]
Can we make the shortcuts on the shortcuts page the same color as the ones on the main page? Aka grey text?
image

**Comment by martinalong** [comment]
image
This is just dummy info right?

**Comment by martinalong** [comment]
image
Loader and the word "saving" are misaligned

**Comment by martinalong** [comment]
I noticed the option + B shortcut (bookmark the page) only works if you already have the extension open. Just typing it normally doesn't seem to trigger a save. Can we make it so it works even without the extension open? Or is there some restriction on it?

**Comment by martinalong** [comment]
image
Can we line these two pieces of text up? "welcome to slate for chrome" should have the same left margin as "slate web extension"

image
And sign in should be bolded like our buttons in the main application

**Comment by martinalong** [comment]
image
When something already exists on slate, hitting "enter" doesn't take you to the app like it's supposed to. Instead, it just tries to save it, then says it already exists:
image

**Comment by martinalong** [comment]
image
The reason your SVGs are unevenly spaced is because they are unevenly sized. This one is wider than the SVG above it (which you seem to be trying to fix with a margin left -3) but that doesn't address the extra width on the right side of it which is now pushing the word "account" to the right.

For all SVGs, the way to standardize them is by putting a set width and height container around it and centering it within that container. Then you don't have to worry about these variations or do hacky things like negative margins. 

If you ever find yourself measuring the positioning of a certain thing and hard coding to adjust it, there's probably a better option. You're always welcome to ask whether there's a "best practice" way to do something, it's one of the best ways to learn--no need to wait till there's a bug to ask questions!

```
//we actually have this style saved in our styles file as STYLES_ICON_CONTAINER
.svgcontainer {
  width: 24px;
  height: 24px;
  display: flex;
  align-items: center;
  justify-content: center;
}

<div classname="svgcontainer">
  <SVG />
</div>
```

**Comment by martinalong** [comment]
image

For this component, one thing you could do is do:
```
display: grid;
grid-template-columns: 50px 1fr auto; //replace the 50px with whatever it actually is
```
This will give the left side icon a fixed width of 50px, the right side will take up however much space it needs, and the middle section will take up whatever is left. This will help you keep each row standard, rather than worrying about aligning each row individually and hoping they match

**Comment by martinalong** [comment]
The option + B bookmark this page still doesn't seem to work

**Comment by martinalong** [comment]
image
You can remove the uploads section since we've decided to no longer have a section to view your upload history

**Comment by martinalong** [comment]
image
Let's also remove this shortcut. Seems unnecessary

**Comment by martinalong** [comment]
image
On the right side there's the grey background for a shortcut, but no shortcut. let's remove the lingering grey background
